lib/robots/sdr_repo/preservation_ingest/transfer_object.rb

```diff
@@ -58,45 +54,30 @@ def prepare_deposit_dir
             deposit_dir.mkpath
           end
         rescue StandardError => e
-          raise(ItemError, "Failed preparation of deposit dir #{deposit_bag_pathname}: #{e.message}")
+          raise ItemError, "Failed preparation of deposit dir #{deposit_bag_pathname}: #{e.message}"
         end
 
         def deposit_dir
           deposit_bag_pathname.parent
         end
 
-        def from_host
-          Settings.transfer_object.from_host
-        end
-
         def from_dir
           Settings.transfer_object.from_dir
         end
 
-        # @see http://en.wikipedia.org/wiki/User:Chdev/tarpipe
-        # ssh user@remotehost "tar -cf - srcdir | tar -C destdir -xf -
-        # Note that symbolic links from /dor/export to /dor/workspace get
-        #  translated into real files by use of --dereference
-        # if command doesn't exit with 0, grabs stdout and stderr and puts them in ruby exception message
         def transfer_bag
-          ssh = "ssh #{from_host} \"tar -C #{from_dir} --dereference -cf - #{bare_druid} \""
-          untar = "tar -C #{deposit_dir} -xf -"
-
-          # If you get an error here it might be a firewall issue between
-          # your preservation robots app and the configured
-          # Settings.transfer.from_host (dor-services-worker)
-          #
-          # This has also been known to fail when the ssh keys in
-          # ~/.ssh/known_hosts includes an invalid public key for the
-          # dor-services-worker, or if a user is required to authorize
-          # the addition of the new public key.
-
-          Open3.pipeline_r(ssh, untar) do |last_stdout, wait_threads|
-            stdout = last_stdout.read # Blocks until complete
-            raise "Transfering bag for #{druid} to preservation failed. STDOUT = #{stdout}" unless wait_threads.map(&:value).all?(&:success?)
+          Open3.popen2e(transfer_command) do |_stdin, stdout_and_stderr, wait_thr|
+            output = stdout_and_stderr.read
+            status = wait_thr.value
+
+            raise "Transfering bag for #{druid} to preservation failed. STDOUT = #{output}" if status.nil? || !status.success?
           end
         end
 
+        def transfer_command
+          "cp -rL #{File.join(from_dir, bare_druid)} #{deposit_dir}"
```

Comment:
If you're wondering whether this could be done in Ruby, `FileUtils.cp_r` only allows for dereferencing the root of what's copied; it does not recursively dereference symlinks, which was the prior behavior. Alas, a `FileUtils`-based approach would have been easier to stub out and test. Lines 70-73 are thus uncovered because the RSpec mechanics to test the block sent to `Open3.popen2e` are tricky and I'm not sure covering these lines provides much value (but am open to being convinced).

Reply:
I think some of this should be in a comment @jmartin-sul